File: src/main/webapp/WEB-INF/static/core/service/ClassificacaoService.js
/**
 * 
 */
app.factory('ClassificacaoService', ['$http', '$q',
                                    function($http, $q) {
	
	var _serverURL = function(url) {
		return SERVER_APP + '/classificacao' + url;
	};
	
	var _save = function(classificacao) {
		return $http.post(_serverURL('/save'), classificacao);
	};

	/**
	 * Busca classificacoes por grupo
	 */
	var _getClassificacoesByGrupo = function(grupo) {
		return $http.get(_serverURL('/get/list/by/grupo/'+grupo.id));
	};

	/**
	 * Busca classificacoes por edicao
	 */
	var _getClassificacoesByEdicao = function(edicao) {
		return $http.get(_serverURL('/get/list/by/edicao/'+edicao.id));
	};
	
	/**
	 * pega a classificacao por edicao e time
	 */
	var _getByEdicaoAndTime = function(edicao, time) {
		return $http.get(_serverURL('/get/list/by/edicao/'+edicao.id+'/time/'+time.id));
	};	
	
	/**
	 * Excluir Classificacao
	 */
	var _deleteClassificacao = function(classificacao) {
		return $http.post(_serverURL('/delete/'+classificacao.id));
	}
	
	
	return {
		
		save : _save,
		
		deleteClassificacao : _deleteClassificacao,
		
		getClassificacoesByGrupo : _getClassificacoesByGrupo,
		
		getClassificacoesByEdicao : _getClassificacoesByEdicao,
		
		getByEdicaoAndTime : _getByEdicaoAndTime
		
	};
	
}]);

File: src/main/java/com/diegolirio/jcampeonato/service/ClassificacaoService.java
package com.diegolirio.jcampeonato.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.diegolirio.jcampeonato.dao.ClassificacaoDao;
import com.diegolirio.jcampeonato.model.Classificacao;
import com.diegolirio.jcampeonato.model.Edicao;
import com.diegolirio.jcampeonato.model.Grupo;
import com.diegolirio.jcampeonato.model.Time;

@Service("classificacaoService")
public class ClassificacaoService extends AbstractGenericService<Classificacao> {

	@Autowired
	private ClassificacaoDao classificacaoDao;

	public List<Classificacao> getClassificacoesByGrupo(Grupo grupo) {
		return this.classificacaoDao.getClassificacoesByGrupo(grupo);
	}

	public List<Classificacao> getClassificacoesByEdicao(Edicao edicao) {
		return this.classificacaoDao.getClassificacoesByEdicao(edicao);
	}
	
	public Classificacao getByEdicaoAndTime(Edicao edicao, Time time) {
		return this.classificacaoDao.getByEdicaoAndTime(edicao, time);
	}

}
